astropy/tests/helper.py

```diff
@@ -131,7 +131,7 @@ def __exit__(self, *exc_info):
     'ipykernel',
     'IPython',   # deprecation warnings for async and await
     'setuptools'])
```

Comment:
maybe add `re` in here as it looks like the warning is coming from there?

Reply:
Okay, will try. Thanks!

Reply:
I meant the package :)

cancelled the build as it's still failing (and I'm not at all sure my suggestion will work anyway).

Reply:
Ah... okay. But, I wasn't convinced that was the correct solution anyway. I might need to refactor a bit for `_warnings_to_ignore_by_pyver` to also take warning class. 🤔

Reply:
yes I'm always confused whether something needs to go into that section or the next one. And it's just full of magic anyway, so 💯 for the refactor